src/opm/output/eclipse/AggregateUDQData.cpp

```diff
@@ -249,18 +359,24 @@ const std::vector<int> Opm::RestartIO::Helpers::igphData::ig_phase(const Opm::Sc
                                                                    const std::vector<int>& inteHead
                                                                   )
 {
+    std::vector<const Opm::Group2*> curGroups(ngmaxz(inteHead), nullptr);
+    for (const auto& group_name : sched.groupNames(simStep)) {
+        const auto& group = sched.getGroup2(group_name, simStep);
+        int ind = (group.name() == "FIELD")
+            ? ngmaxz(inteHead)-1 : group.insert_index()-1;
+        curGroups[ind] = std::addressof(group);
+    }
     std::vector<int> inj_phase(ngmaxz(inteHead), 0);
-
-    for (const auto& gname : sched.groupNames(simStep)) {
-        const auto& group = sched.getGroup2(gname, simStep);
-        if (group.isInjectionGroup()) {
-            //auto phase = group.getInjectionPhase();
-            auto phase = Opm::Phase::OIL;
-            if ( phase == Opm::Phase::OIL   ) inj_phase[group.insert_index()] = 1;
-            if ( phase == Opm::Phase::WATER ) inj_phase[group.insert_index()] = 2;
-            if ( phase == Opm::Phase::GAS   ) inj_phase[group.insert_index()] = 3;
+    for (std::size_t ind = 0; ind < curGroups.size(); ind++) {
+        if (curGroups[ind] != nullptr) {
```

Comment:
Can not be `nullptr` - the value returned from `Schedule::getGroup2()`will always be valid - if not an exception will be thrown in the `Schedule::getGroup2()`